### PR TITLE
Add @Nullable to ValueIn.object arg

### PR DESCRIPTION
**Pull request by glukos**

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/wire/ValueIn.java [hunk 1]
```diff
@@ -513,8 +513,12 @@ default <E extends Enum<E>, T> WireIn asEnum(Class<E> eClass, T t, @NotNull BiCo
         return wireIn();
     }
 
+    /**
+     * @param clazz Expected object type. <code>null</code> can be passed to request proxy marshallable tuple
+     *              if {@link Wires#GENERATE_TUPLES} is enabled.
+     */
     @Nullable
-    default <E> E object(@NotNull Class<E> clazz) {
+    default <E> E object(@Nullable Class<E> clazz) {
```
Comment (alamar) [inline src/main/java/net/openhft/chronicle/wire/ValueIn.java:521]: Perhaps net.openhft.chronicle.wire.JSONWire.JSONValueIn#object(java.lang.Class<E>) needs updating also.

Comment (glukos) [reply src/main/java/net/openhft/chronicle/wire/ValueIn.java:521]: Thanks, fixed

**Comment by alamar** [comment]
Please see SonarCloud - looks like we also need to take care of parseType()

**Comment by glukos** [comment]
@alamar Thanks again, I've missed Sonar report.
Non-null clazz was required by `parseType` to check inheritance - now `null` value is treated like `Object` that inherits everything.

**Review by alamar — APPROVED** [review]

**Pull request merged**